packages/portfolio-contract/test/portfolio-open.test.ts

```diff
@@ -0,0 +1,164 @@
+              log({ _cap: addr.value, _method: 'transfer', address, amount });
+              const { transfer: err } = errs;
+              if (err) throw err;
+            },
+            async executeEncodedTx(msgs) {
+              log({ _cap: addr.value, _method: 'executeEncodedTx', msgs });
+              const { executeEncodedTx: err } = errs;
+              if (err) throw err;
+              return harden(msgs.map(_ => ({})));
+            },
+          });
+        },
+      });
+    },
+  }) as unknown as Orchestrator;
+
+  const zone = makeHeapZone();
+  const makePortfolioKit = preparePortfolioKit(zone);
+  const zoeTools = harden({
+    async localTransfer(sourceSeat, localAccount, amounts) {
+      log({ _method: 'localTransfer', sourceSeat, localAccount, amounts });
+      const { localTransfer: err } = errs;
+      if (err) throw err;
+    },
+    async withdrawToSeat(localAccount, destSeat, amounts) {
+      log({ _method: 'withdrawToSeat', localAccount, destSeat, amounts });
+    },
+  }) as GuestInterface<ZoeTools>;
+
+  const { brand: USDC } = makeIssuerKit('USDC');
+  const proposal: Proposal = harden({
+    give: { USDN: AmountMath.make(USDC, 100n) },
+  });
+  let hasExited = false;
+  const seat = {
+    getProposal: () => proposal,
+    hasExited: () => hasExited,
+    exit(completion) {
+      if (hasExited) throw Error('already exited');
+      log({ _cap: 'seat', _method: 'exit', completion });
+      hasExited = true;
+    },
+    fail(reason) {
+      if (hasExited) throw Error('already exited');
+      log({ _cap: 'seat', _method: 'fail', reason });
+      hasExited = true;
+    },
+  } as ZCFSeat;
+
+  const inertSubscriber = {} as ResolvedPublicTopic<never>['subscriber'];
+
+  return { orch, log: buf, makePortfolioKit, zoeTools, seat, inertSubscriber };
+};
+
+test('open portfolio', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks();
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
+  ]);
+});
+
+test('handle failure in localTransfer from seat to local account', async t => {
+  const { orch, log, makePortfolioKit, zoeTools, seat, inertSubscriber } =
+    mocks({ localTransfer: Error('localTransfer from seat failed') });
+
+  const localP = makeLocalAccount(orch, {});
+  const actual = await openPortfolio(
+    orch,
+    { zoeTools, makePortfolioKit, inertSubscriber },
+    seat,
+    undefined,
+    localP,
+  );
+  t.snapshot(log, 'call log');
+  t.is(passStyleOf(actual.invitationMakers), 'remotable');
+  t.like(actual.publicTopics, [
+    { description: 'USDN ICA', storagePath: 'cosmos:noble-3:noble11028' },
+  ]);
```

Comment:
_question_

Just to confirm, did we not observe anything different if `localTransfer` failed compared to the happy path? Would `localTransfer` ever fail?

Reply:
The differences are in the call log in the snapshot.

Reply:
I suppose I could add some test assertions about the log too